govcd/catalog.go

```diff
@@ -864,6 +864,21 @@ func (cat *Catalog) GetCatalogItemByName(catalogItemName string, refresh bool) (
 	return nil, ErrorEntityNotFound
 }
 
+// GetVAppTemplateByName finds a VAppTemplate by Name
+// On success, returns a pointer to the VAppTemplate structure and a nil error
+// On failure, returns a nil pointer and an error
+func (cat *Catalog) GetVAppTemplateByName(vAppTemplateName string, refresh bool) (*VAppTemplate, error) {
```

Comment:
What happens if the catalog item name and the vApp template name are different?

Reply:
Thanks for the test. This behavior and the impossibility of un-matching names should be documented.
As for deprecating catalog item, please hold on to it. I'd like to discuss the whole plan first.

Reply:
I've added a few lines to `VAppTemplate.Update` to state this behaviour.